pkg/controller/keystone/keystone_controller.go

```diff
@@ -296,6 +310,30 @@ func (r *ReconcileKeystone) updateStatus(
 	return r.client.Status().Update(context.Background(), k)
 }
 
+func (r *ReconcileKeystone) externalKeystoneReady(k *contrail.Keystone, adminPasswordSecret string) (bool, error) {
+	keystoneClient, err := keystone.NewClient(r.client, r.scheme, r.restConfig, k)
+	if err != nil {
+		return false, err
+	}
+	_, err = keystoneClient.PostAuthTokens("admin", adminPasswordSecret, "admin")
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get keystone token: %v", err)
+	}
+	return true, nil
+}
+
+func (r *ReconcileKeystone) updateStatusWithExternalKeystone(
+	k *contrail.Keystone,
+) error {
+	k.Status = contrail.KeystoneStatus{}
+	k.Status.Active = true
+	k.Status.External = true
+	k.Status.Port = k.Spec.ServiceConfiguration.ListenPort
+	k.Status.ClusterIP = k.Spec.ServiceConfiguration.ExternalAddress
```

Comment:
maybe if right now we will support external keystone this field in status should have different name than ClusterIP since it can be some externalIP or even domain name if I understood correctly, so please rename it to Endpoint or sth.

Reply:
Renamed to Endpoint, also for internal variables changed IP references to more general 'address'.